src/test/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherITest.java

```diff
@@ -76,18 +95,59 @@ void shouldPublishGitHubCheckRunCorrectly() throws IOException {
                                         .withEndColumn(30)
                                         .withTitle("Hello GitHub Checks API")
                                         .withRawDetails("a simple echo command")
-                                        .build()
-                        ))
+                                        .build()))
                         .withImages(Collections.singletonList(
                                 new ChecksImage("Jenkins",
                                         "https://ci.jenkins.io/static/cd5757a8/images/jenkins-header-logo-v2.svg",
-                                        "Jenkins Symbol")
-                        ))
+                                        "Jenkins Symbol")))
                         .build())
                 .withActions(Collections.singletonList(
                         new ChecksAction("re-run", "re-run Jenkins build", "#0")))
                 .build();
 
+        new GitHubChecksPublisher(createGitHubChecksContextWithGitHubSCM(), jenkinsRule.createTaskListener(),
+                wireMockRule.baseUrl())
+                .publish(details);
+    }
+
+    /**
+     * If exception happens when publishing checks, it should output all parameters of the check to the system log.
+     */
+    @Issue("issue-20")
+    @Test
+    public void shouldLogChecksParametersIfExceptionHappensWhenPublishChecks() {
+        loggerRule.record(GitHubChecksPublisher.class.getName(), Level.WARNING).capture(1);
+
+        ChecksDetails details = new ChecksDetailsBuilder()
+                .withName("Jenkins")
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(ChecksConclusion.SUCCESS)
+                .withOutput(new ChecksOutputBuilder()
+                        .withTitle("Jenkins Check")
+                        .withSummary("# A Successful Build")
+                        .withAnnotations(Collections.singletonList(
+                                new ChecksAnnotationBuilder()
+                                        .withPath("Jenkinsfile")
+                                        .withStartLine(1)
+                                        .withEndLine(2)
+                                        .withStartColumn(0)
+                                        .withEndColumn(20)
+                                        .withAnnotationLevel(ChecksAnnotationLevel.WARNING)
+                                        .withMessage("say hello to Jenkins")
+                                        .build()))
+                        .build())
+                .build();
+
+        new GitHubChecksPublisher(createGitHubChecksContextWithGitHubSCM(), jenkinsRule.createTaskListener(),
+                wireMockRule.baseUrl())
+                .publish(details);
+
+        assertThat(loggerRule.getRecords().size()).isEqualTo(1);
+        assertThat(loggerRule.getMessages()).contains("Failed Publishing GitHub checks: "
```

Comment:
Can you assert the exact value please not rely on the to string it’s impossible to see if it’s actually working here or what the output is

Reply:
Or at least a contains string

Reply:
Yes, something like `.contains("x = valueX").contains("y = valueY")`